src/main/java/org/semver4j/internal/Coerce.java

```diff
@@ -4,30 +4,30 @@
 import org.jetbrains.annotations.Nullable;
 
 import java.util.Locale;
+import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 import static java.lang.String.format;
 import static java.util.regex.Pattern.compile;
-import static org.semver4j.internal.Tokenizers.COERCE;
 
 public class Coerce {
     @NotNull
-    private static final Pattern pattern = compile(COERCE);
+    private static final Pattern PATTERN = compile(
+        "(^|\\D)(\\d{1,16})(?:\\.(\\d{1,16}))?(?:\\.(\\d{1,16}))?(?:$|\\D)"
+    );
+
+    private Coerce() {
+    }
 
     @Nullable
     public static String coerce(@NotNull final String version) {
-        Matcher matcher = pattern.matcher(version);
+        Matcher matcher = PATTERN.matcher(version);
 
         if (matcher.find()) {
-            String group1 = matcher.group(0);
-            String group2 = matcher.group(1);
             String group3 = matcher.group(2);
-            String group4 = matcher.group(3);
-            String group5 = matcher.group(4);
-
-            group4 = group4 != null ? group4 : "0";
-            group5 = group5 != null ? group5 : "0";
+            String group4 = Optional.ofNullable(matcher.group(3)).orElse("0");
+            String group5 = Optional.ofNullable(matcher.group(4)).orElse("0");
```

Comment:
If you removed these unused groups, right now variable names like `group3`, `group4`, and `group5` seem a little odd. Perhaps they should be named:

* `group3` -> `coercedMajor`
* `group4` -> `coercedMinor`
* `group5` -> `coercedPatch`

or something similar.